packages/babel-core/test/api.js

```diff
@@ -168,6 +182,30 @@ describe("api", function() {
     expect(options).toEqual({ babelrc: false });
   });
 
+  it("transformFromAst should not mutate the AST", function() {
+    const program = "const identifier = 1";
+    const node = parse(program);
+    const { code } = transformFromAst(node, program, {
+      plugins: [
+        function() {
+          return {
+            visitor: {
+              Identifier: function(path) {
+                path.node.name = "replaced";
+              },
+            },
+          };
+        },
+      ],
+    });
+
+    expect(code).toBe("const replaced = 1;");
+    expect(node.program.body[0].declarations[0].id.name).toBe(
```

Comment:
Since there was a discussion on deep cloning the test (or a new test) should probably reflect that, like check the the other metadata fields aren't changed?

Reply:
That would be nice, yes. Any ideas on how to force a metadata change? I'm outputing the AST for this transform and it doesn't seem to change any metadata fields...

Reply:
I can add something like that if you folks think it would be useful. I am not really in a good position to evaluate that, though. Do people usually sprinkle such metadata fields into their ASTs?

Reply:
Actually this works fine! thanks!